liquidation-monitor/src/agent.js

```diff
@@ -111,6 +129,18 @@ function provideInitialize(data) {
     data.newAccounts = []; // New account from transaction events
     data.totalNewAccounts = 0;
 
+    // Calculate the next report time.
+    let reportingHour = 12; // reset the alerts on this UTC hour. Set to noon UTC
+    reportingHour %= 24; // Modulo to keep it within 24 hours.
+    // Convert JS time in milliseconds to UTC time in seconds.
+    const now = (new Date().getTime()) / 1000;
+    // Calculate midnight of yesterday 00:00:00
+    const oneDay = 86400;
+    const today = now - (now % oneDay); // now minus seconds elapsed since midnight.
+    const oneHour = 3600;
+    data.nextAlertTime = today + (reportingHour * oneHour);
+    data.alertedAccounts = []; // Accounts alerted in the last 24 hours.
```

Comment:
It's a feature. In order to to have groups of alerts occur at the same time of day regardless of when the bot is launched, this logic is needed. If I only wanted  alerts to repeat at UTC midnight, then it can be reduced down to ```js
const now = (new Date().getTime()) / 1000;
const oneDay = 86400;
data.nextAlertTime = now - (now % oneDay);```

Having the alert group repeat at noon seemed better than 4:53 pm, or whenever the bot was started.

Reply:
I agree with simplifying this, but I suggest doing it in a slightly different way.

I think that Date().getTime() depends upon the system clock, which means it probably shouldn't be relied upon for synchronization.

Because these have access to blockchain data, I think grabbing the current block (yes, I realize it's in the initialization function) might be a better source.

The following suggestion will get the latest block timestamp, subtract whatever fraction of a day has elapsed, then add a day to it for the next alert time.
```
    const latestBlock = data.provider.getBlock('latest');
    data.nextAlertTime = latestBlock.timestamp - (latestBlock.timestamp % 86400) + 86400;
```
Because Scan Nodes may receive and start this Bot at _slightly_ different times, the removal of any fraction of a day should baseline their starting times to a day boundary.  Depending upon when you deploy the Bot, some Scan Nodes may start the Bot just before a 24-hour boundary and some may start it just after, but they should all end up synchronized on the same day boundaries once blocks start getting processed.

For this to work, you would need to only increment the timestamp by exactly 86400 so that all Bots will alert on the same day boundaries.

Reply:
The original consideration is that 2 computers might have varying times IE: 1653609699 and 1653609650. But they would both calculate Midnight of May 27th to be 1653609600. That value would be stored in alert time. The following checks would be based on Eth blocktime vs the alert time, which would put all bots in sync. The limitation to this approach is nodes with more than 24 hour time difference would be on different days.

I concur, that blocktime as the starting point would be most accurate. Fixed in commit 31fe567753617fed31cafbda628011d5f686e0ac